06.bowling_object/frame.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require_relative 'shot'
+
+class Frame
+  LAST_FRAME = 9
+  MAX_SHOTS_IN_LAST_FRAME = 3
+  MAX_SHOTS_IN_REGULAR_FRAME = 2
+
+  attr_reader :scores
+
+  def initialize(frame_idx)
+    @scores = []
+    @frame_idx = frame_idx
+  end
+
+  def add_shot(marks)
+    marks.each do |mark|
+      @scores << Shot.new(mark)
+      break if complete?
+    end
+    @scores.size
+  end
+
+  def score(frames)
+    @frame_idx != LAST_FRAME ? @scores.sum(&:score) + bonus_score(frames) : @scores.sum(&:score)
+  end
+
+  private
+
+  def complete?
+    if @frame_idx != LAST_FRAME
```

Comment:
```suggestion
    if @frame_idx == LAST_FRAME
```

否定条件よりも肯定条件を前に持ってきた方が読みやすいと思います。

もしくはこうするとか

```suggestion
    if @frame_idx < LAST_FRAME
```

Reply:
否定条件を置かない方が読みやすいですね。